petab/conditions.py

```diff
@@ -24,20 +24,12 @@ def get_condition_df(
 
     if isinstance(condition_file, str):
         condition_file = pd.read_csv(condition_file, sep='\t',
-                                     float_precision='round_trip')
+                                     float_precision='round_trip',
+                                     index_col=CONDITION_ID)
 
     lint.assert_no_leading_trailing_whitespace(
         condition_file.columns.values, "condition")
 
-    if not isinstance(condition_file.index, pd.RangeIndex):
```

Comment:
I think it was requested by somebody (@yannikschaelte ?) to have this function set the index correctly if not already done. Feels a bit funny, so okay for me to drop.

Reply:
> Ah, I see. Okay, I think we want to have `index_col=CONDITION_ID` there.

???

Reply:
@dweindl Nope, not aware of these lines even ...

Reply:
Ah, this becomes relevant though when DataFrames are directly passed to this function and not read from file. The question is just whether DataFrame shouldn't have been sanitized before.

Reply:
Right. In my opinion they should be in the proper shape and just allowed to be passed through for convenience. But there were other opinions, but don't remember by whom. Okay for me to change.